libr/anal/esil.c

```diff
@@ -587,6 +587,55 @@ static bool esil_signext(RAnalEsil *esil) {
 	return r_anal_esil_pushnum (esil, ((src ^ m) - m));
 }
 
+// sign extension assignement
+// example : > "ae 0x81,a0,="
+//           > "ae 8,a0,~="   ( <src bit width>,register,~= )
+// output  : > ar a0
+//           0xffffff81
+static bool esil_signexteq(RAnalEsil *esil) {
+	ut64 src, dst;
+
+	char *p_src = r_anal_esil_pop (esil);
```

Comment:
Ideally it will be good to share most code with  `esil_signext`, can you refactor this? we should be able to describe operations in a functional way without having to go into much details in the code. X= operations must be calling internally the X operation and then the = operation for example.

Reply:
I refactored to share code but doing that in a functional way is not obvious to me.